product_sequence/models/product_product.py

```diff
@@ -23,18 +23,15 @@ def create(self, vals):
         if 'default_code' not in vals or vals['default_code'] == '/':
             categ_id = vals.get("categ_id")
             template_id = vals.get("product_tmpl_id")
-            categ = sequence = False
+            categ = self.env['product.category']
             if categ_id:
                 # Created as a product.product
                 categ = self.env['product.category'].browse(categ_id)
             elif template_id:
                 # Created from a product.template
                 template = self.env["product.template"].browse(template_id)
                 categ = template.categ_id
-            if categ:
-                sequence = categ.sequence_id
-            if not sequence:
-                sequence = self.env.ref('product_sequence.seq_product_auto')
+            sequence = categ.get_sequence_id()
```

Comment:
Maybe you can keep the If clause, and call you function there.

Reply:
Wait. Actually "categ_id" of product is required field. Can you create a product without category?